var/spack/repos/builtin/packages/warpx/package.py

```diff
@@ -69,6 +70,11 @@ class Warpx(CMakePackage):
     depends_on('blaspp', when='+psatd dims=rz')
     depends_on('blaspp +cuda', when='+psatd dims=rz compute=cuda')
     depends_on('boost@1.66.0: +math', when='+qedtablegen')
+
+    # TODO: replace this with an explicit list of components of Boost,
+    # for instance depends_on('boost +filesystem')
+    # See https://github.com/spack/spack/pull/22303 for reference
+    depends_on(Boost.with_default_variants, when='+qedtablegen')
```

Comment:
Given the comment below, should this read:
```suggestion
    depends_on('boost@1.66.0: cxxstd=14', when='+qedtablegen')
```

(I think spec-ing `cxxstd=14` is a good idea since we build with C++14. The math libs are header-only.)